src/main/kotlin/interpreter/pyobject/types/PyBytesType.kt

```diff
@@ -0,0 +1,45 @@
+package green.sailor.kython.interpreter.pyobject.types
+
+import green.sailor.kython.interpreter.Exceptions
+import green.sailor.kython.interpreter.functions.PyBuiltinFunction
+import green.sailor.kython.interpreter.iface.ArgType
+import green.sailor.kython.interpreter.iface.PyCallableSignature
+import green.sailor.kython.interpreter.pyobject.*
+import java.lang.IndexOutOfBoundsException
+
+object PyBytesType : PyType("bytes") {
+    override fun newInstance(kwargs: Map<String, PyObject>): PyObject {
+        val value = kwargs["value"] ?: error("Built-in signature mismatch")
+        when (value) {
+            is PyString -> {
+                return PyBytes(value.wrappedString.toByteArray())
+            }
+            else -> error("Not yet supported")
+        }
+    }
+
+    override val signature: PyCallableSignature by lazy {
+        PyCallableSignature(
+            "value" to ArgType.POSITIONAL
+        )
+    }
+
+    /** bytes.__getitem__ */
+    val pyBytesIndex = PyBuiltinFunction.wrap("__getitem__", PyCallableSignature("index" to ArgType.POSITIONAL)) {
+        val self = it["self"]!!.cast<PyBytes>()
```

Comment:
These should error out if the non-null assertion fails

Reply:
This is copied from `PyStringType`, and it can be assumed that this is always a valid cast.

Reply:
The non-null assertion is fine because it'll never error out.